File: 프로그래머스/크레인 인형뽑기 게임/crain.py
def solution(board, moves):
    answer =0
    arr = []
    for i in moves:
        x = i-1;
        for y in range(len(board)):
            if board[y][x] != 0:
                if not (not arr) and board[y][x] == arr[-1]:
                    arr.pop()
                    answer+=2
                else :
                    arr.append(board[y][x])
                board[y][x] = 0
                break
    return answer
if __name__ == "__main__" :
    b = [[0,0,0,0,0],[0,0,1,0,3],[0,2,5,0,1],[4,2,4,4,2],[3,5,1,3,1]]
    m = [1,5,3,5,1,2,1,4]
    a= solution(b,m)

File: 프로그래머스/크레인 인형뽑기 게임/crain.java
import java.util.*;
class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> s = new Stack<>();
        for(int x=0; x< moves.length; x++){
            for(int y=0;y<board.length;y++){
                if(board[y][moves[x]-1] !=0){
                    if(!s.empty() && s.peek()==board[y][moves[x]-1]){
                        s.pop();
                        answer +=2;
                    }
                    else
                        s.push(board[y][moves[x]-1]);
                    board[y][moves[x]-1] =0;
                    break;
                }
            }
        }
        return answer;
    }
}
public class test{
    public static void main(String[] args){

    }
}
